sdk/textanalytics/azure-ai-textanalytics/azure/ai/textanalytics/_response_handlers.py

```diff
@@ -41,6 +41,11 @@ def process_batch_error(error):
     """Raise detailed error message.
     """
     raise_error = HttpResponseError
+    if error.status_code == 200:
+        # this is the case where we manually throw the Too Many Documents error
+        # we don't want to raise an error based on the error.response
+        error.response.status_code = 400
+        raise HttpResponseError(response=error.response, message=error.message)
```

Comment:
just realizing, because we're not passing `error_format=CSODataV4Format`, if the user catches the error they won't be able to access the error code like this since it won't get set:

```
try:
    r = analyze_sentiment(docs)
except HttpResponseError as e:
   print(e.error.code)
```

what if, in the CSODataV4Format class, we check for `errors` since full response is being returned (above line 34 in try block):

```
if "errors" in odata_error:
    odata_error = odata_error["errors"][0]
```

Reply:
That sounds good, let me play around with that. Thanks @kristapratico